### ISSUE TITLE
Unwanted dataset shuffling during evaluation with random holdout in backed mode

### ISSUE THREAD

**Issue by le-ander** [theislab/sfaira#6]
Just to keep track of the issue here.

We suspect that the following lines of code are the culprit:
https://github.com/theislab/sfaira/blob/27b84f431e224ecafbfda6774943ffeaf7b07a74/sfaira/estimators/keras.py#L598-L600

Similarly, for celltype models:
https://github.com/theislab/sfaira/blob/27b84f431e224ecafbfda6774943ffeaf7b07a74/sfaira/estimators/keras.py#L1077-L1079

This could also cause an unrelated problem:
https://github.com/theislab/sfaira/blob/27b84f431e224ecafbfda6774943ffeaf7b07a74/sfaira/train/train_model.py#L293

Thanks for investigating, Abdul!
